scripts/release.sh

```diff
@@ -133,6 +133,25 @@ git push "${REPOSITORY}" "${RELEASE_TAG}" ${GIT_DRYRUN:-}
 echo "Deploying release to maven central"
 mvn deploy -Prelease -DskipTests=true -DreleaseSigningKey="${GPG_KEY}" -DprocessAllModules=true ${MVN_DEPLOY_DRYRUN:-}
 
+echo "Creating binary distribution"
+mvn package -Pdist -DskipTests=true
+
+if ! command -v gh &> /dev/null
+then
+    echo "gh command could not be found. Please create a release by hand https://github.com/kroxylicious/kroxylicious/releases/new"
+else
+  echo "Creating GitHub release with binary distribution"
+  ORIGINAL_GH_DEFAULT_REPO=$(gh repo set-default -v | (grep -v 'no default repository' || true))
+  gh repo set-default $(git remote get-url ${REPOSITORY})
+  # create GitHub release via CLI https://cli.github.com/manual/gh_release_create
+  if [[ "${DRY_RUN:-false}" == true ]]; then
+    gh release create "${RELEASE_TAG}" ./kroxylicious-*/target/kroxylicious-*-bin.* --title "${RELEASE_TAG}" --notes-file "CHANGELOG.md" --draft
+    echo "Draft release created. This must be manually published."
+  else
+    gh release create "${RELEASE_TAG}" ./kroxylicious-*/target/kroxylicious-*-bin.* --title "${RELEASE_TAG}" --notes-file "CHANGELOG.md"
```

Comment:
`--notes-file "CHANGELOG.md"`

does it have some smarts around the notes-file or does it just dump the entire contents as the notes for the release?

Reply:
Just a dumb copy for now, which I think is fine for this PR. 

Thinking about it maybe we change things so the `changelog.md` represents the current SNAPSHOT release and every time we create a release we create a `changelog-Major.Minor.Patch.md` file?

Reply:
If we were to have `CHANGELOG.md` only contain the changes for the current version (i.e. at the tag for `v0.4.0` it contains all changes between `v0.3.0` and that commit, and on main it contains all changes since the last release) then I think that would achieve what we want. Then if anyone wants to see the changelog for a previous version, they can either go look at the releases page or checkout the commit for that release (which would have a `CHANGELOG.md` containing only that release's changes).